### PR TITLE
Fix 'Color Color' completion logic

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/70403

### CODE REVIEW

File: src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationServiceRunner.cs [hunk 1]
```diff
@@ -17,6 +17,7 @@
 using Microsoft.CodeAnalysis.PooledObjects;
```
Comment (CyrusNajmabadi) [inline src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationServiceRunner.cs:17]: review with whitespace off.

**Comment by CyrusNajmabadi** [comment]
@genlu  ptal.

File: src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationServiceRunner.cs [hunk 2]
```diff
@@ -531,145 +520,208 @@ private RecommendedSymbols GetSymbolsOffOfConditionalReceiver(ExpressionSyntax o
...
+                var semanticModel = _context.SemanticModel;
+                var symbol = leftHandBinding.GetAnySymbol();
 
-                // If the thing on the left is a method name identifier, we shouldn't show anything.
-                if (symbol.Kind is SymbolKind.Method &&
-                    originalExpressionKind is SyntaxKind.IdentifierName or SyntaxKind.GenericName)
+                // If the symbol is currently bound as a named type, try to bind it as an instance.  Conversely, if it's
+                // bound as an instance, try to bind it as a named type.
+                INamedTypeSymbol? instanceType, staticType;
+                if (symbol is INamedTypeSymbol namedType)
```
Comment (genlu) [inline src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationServiceRunner.cs:554]: OK, so the issue is caused by the node being bound as a type, which our previous logic didn't handle?

**Review by genlu — APPROVED** [review]
:shipit:

File: src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationServiceRunner.cs (hunk 2 above)
Comment (CyrusNajmabadi) [reply src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationServiceRunner.cs:554]: the opposite :).  We bound it as an instance, so we didn't show the static member.  

The prior logic said "if we bound it as a type, also try as an instance".  Now we have the reverse "if we bound it as an instance, also try it as a type".

**Pull request merged**